File: backend/app/models/reservation.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Reservation(Base):
    """
    会議室予約情報を管理するDBモデル。

    ユーザー、会議室、予約時間、予約目的、有効状態を管理する。
    """

    __tablename__ = "reservations"

    id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        index=True,
    )
    user_id: Mapped[int] = mapped_column(
        ForeignKey("users.id"),
        nullable=False,
        index=True,
    )
    meeting_room_id: Mapped[int] = mapped_column(
        ForeignKey("meeting_rooms.id"),
        nullable=False,
        index=True,
    )
    title: Mapped[str] = mapped_column(
        String(100),
        nullable=False,
    )
    start_at: Mapped[datetime] = mapped_column(
        DateTime,
        nullable=False,
        index=True,
    )
    end_at: Mapped[datetime] = mapped_column(
        DateTime,
        nullable=False,
        index=True,
    )
    is_active: Mapped[bool] = mapped_column(
        Boolean,
        nullable=False,
        default=True,
    )

File: frontend/src/types/reservation.ts
export type Reservation = {
  id: number;
  user_id: number;
  meeting_room_id: number;
  title: string;
  start_at: string;
  end_at: string;
  is_active: boolean;
};

export type ReservationCreateRequest = {
  user_id: number;
  meeting_room_id: number;
  title: string;
  start_at: string;
  end_at: string;
};

export type ReservationUpdateRequest = {
  user_id: number;
  meeting_room_id: number;
  title: string;
  start_at: string;
  end_at: string;
  is_active: boolean;
};